x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/frozen/action/TransportFreezeIndexAction.java

```diff
@@ -133,6 +136,20 @@ private void toggleFrozenSettings(final Index[] concreteIndices, final FreezeReq
                 })) {
             @Override
             public ClusterState execute(ClusterState currentState) {
+                List<String> writeIndices = new ArrayList<>();
```

Comment:
I didn't add a unit test for this, because currently for this code no unit tests exist.
The `Cannot freeze write index for data stream` yaml test does test this.